src/ansys/sound/core/sound_composer/source_harmonics_two_parameters.py

```diff
@@ -0,0 +1,400 @@
+                f"\t\tMax: {self.control2.control.data.max()}\n"
+                f"\t\tDuration: "
+                f"{self.control2.control.time_freq_support.time_frequencies.data[-1]} s"
+            )
+        else:
+            str_control_rpm = "Not set"
+            str_control2 = "Not set"
+
+        return (
+            f"Harmonics source with two parameters: {str_source}\n"
+            f"Source control:\n"
+            f"\tControl 1: {str_control_rpm}\n"
+            f"\tControl 2: {str_control2}"
+        )
+
+    @property
+    def control_rpm(self) -> SourceControlTime:
+        """First source control, that is, RPM, for harmonics source with two parameters.
+
+        Contains the RPM values over time.
+        """
+        return self.__control_rpm
+
+    @control_rpm.setter
+    def control_rpm(self, control: SourceControlTime):
+        """Set the source control."""
+        if not (isinstance(control, SourceControlTime) or control is None):
+            raise PyAnsysSoundException(
+                "Specified RPM source control object must be of type ``SourceControlTime``."
+            )
+        self.__control_rpm = control
+
+    @property
+    def control2(self) -> SourceControlTime:
+        """Second source control for harmonics source with two parameters.
+
+        Contains the second control parameter values over time.
+        """
+        return self.__control2
+
+    @control2.setter
+    def control2(self, control: SourceControlTime):
+        """Set the source control."""
+        if not (isinstance(control, SourceControlTime) or control is None):
+            raise PyAnsysSoundException(
+                "Specified second source control object must be of type ``SourceControlTime``."
+            )
+        self.__control2 = control
+
+    @property
+    def source_harmonics_two_parameters(self) -> FieldsContainer:
+        """Source data for harmonics source with two parameters.
+
+        The harmonics source with two parameters data consists of a series of orders whose levels
+        depend on the values of two parameters, the first of which is always RPM.
+        """
+        return self.__source_harmonics_two_parameters
+
+    @source_harmonics_two_parameters.setter
+    def source_harmonics_two_parameters(self, source: FieldsContainer):
+        """Set the harmonics source with two parameters data, from a DPF fields container."""
+        if source is not None:
+            if not isinstance(source, FieldsContainer):
+                raise PyAnsysSoundException(
+                    "Specified harmonics source with two parameters must be provided as a DPF "
+                    "fields container."
+                )
+
+            if len(source) < 1:
+                raise PyAnsysSoundException(
+                    "Specified harmonics source with two parameters must contain at least one "
+                    "order."
+                )
+
+            for spectrum in source:
```

Comment:
@ansaminard Replace this for by:

```
nb_orders = len(self.source_harmonics_two_parameters[0].time_freq_support.time_frequencies.data) # nb orders
if (nb_orders < 1)
display error

for f_order in source:
if len(f_order.data) != nb_orders:
raise PyAnsysSoundException(
"Each set of order in the specified harmonics source with two parameters must contain the same number of orders"
)
```

By the way, there were "spectrum" instead of "orders_levels"

Reply:
Done and done
(different tests than suggested, for consistency with harmonics source)